js/main.js

```diff
@@ -748,11 +748,19 @@ class PainterroProc {
           if (
             (evt.keyCode === KEYS.y && evt.ctrlKey) ||
             (evt.keyCode === KEYS.z && evt.ctrlKey && evt.shiftKey)) {
-            this.worklog.redoState();
-            e.preventDefault();
+            if (this.params.userRedo) {
+              this.params.userRedo.call();
+            } else {
+              this.worklog.redoState();
```

Comment:
@AlonaRadchenko cool, but I think when callback is used, we still need to do undo and redo. So lets remove else, and moce code from else to top level